### DISCUSSION TITLE
Unable to upload files via nginx to nextcloud

### DISCUSSION THREAD

**Discussion by Pyroseza** [evertramos/nginx-proxy-automation#268] [Q&A] [answered]
**Describe the bug**
When paired with `docker-nextcloud-letsencrypt` it's not possible to actually upload any files to NextCloud, instead you just get a popup in the top right corner of NextCloud saying: `an unknown error has occurred`.

The problem is not actually with NextCloud, but with Nginx, and the reason I've logged the bug for this repo is that it is not configured in such a way to override the Nginx default value for client max body size of `1 MiB`.

I could make a change locally but it does not persist after upgrades, and I'm sure other people would like this to work nicely.

**To Reproduce**
1. Spin up a blank `nginx-proxy-automation`
2. Spin up a blank `docker-nextcloud-letencrypt` with SSL
3. Log into nextcloud via https
4. Try upload a photo / video or whatever of at least a few MB
5. See error in NextCloud `an unknown error has occurred`

**Expected behavior**
I expected my file to upload to NextCloud without any issues.

**Server info:**
 - Linux release: Centos 7.9
 - Server type: SuperMicro SuperServer
 - Docker version: 20.10.5
 - docker-compose version: 1.26.2

**Logs (please send some logs):**
 - nginx container
```
proxy-web-auto      | 2021/04/04 13:39:52 [error] 61#61: *1 client intended to send too large body: 6228259 bytes, client: <redacted ip address>, server: <redacted url>, request: "PUT /remote.php/webdav/<redacted filename> HTTP/2.0", host: "<redacted url>"
```
 - nextcloud container -> unfortunately it does not log the error as it's caught by nginx

**Additional context**
I found a temp solution for myself and that is to add this line into the nginx template:
`client_max_body_size 1024M;`

![image](https://user-images.githubusercontent.com/8463626/113512299-eca91a80-955b-11eb-8c5f-c13d1d57560b.png)

I'm not proposing this as a solution, but maybe there could be a more configurable way of choosing which containers get this feature added and maybe a way to control the size for each container.

**Comment by evertramos** [comment, marked as answer]
Please check here and let me know if this was helpful:

https://github.com/evertramos/nginx-proxy-automation/discussions/261#discussioncomment-568310

If not you can continue in the discution your question.
